### PR TITLE
Jackson improvements

### PR DESCRIPTION
**Pull request by herau**
refactor all model class to remove useless jackson annotations
Create a Json Utils class to use a shared ObjectMapper ([performance](http://wiki.fasterxml.com/JacksonBestPracticesPerformance))
Optimize json reading
add some unit tests

### CODE REVIEW

**Comment by itchix** [comment]
Wow :)

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/FoodAPIRestClientUsage.java [hunk 1]
```diff
@@ -277,21 +276,14 @@ public void onStart() {
...
-                        }
-                        onAllergensCallback.onAllergensResponse(true);
-                    } catch (InvalidFormatException e) {
-                        e.printStackTrace();
-                        onAllergensCallback.onAllergensResponse(false);
+                    AllergenRestResponse restResponse = JsonUtils.readFor(AllergenRestResponse.class)
+                            .readValue(responseBody);
+
+                    for (Allergen allergen : restResponse.getAllergens()) {
+                        allergen.save();
```
Comment (herau) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/FoodAPIRestClientUsage.java]: @itchix do you know what is the goal of this loop on the list to apply a save method on each element ?

Comment (itchix) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/FoodAPIRestClientUsage.java]: It's to store in the DB all allergens from the API, and then I can use it in this view : AlertUserFragment

Comment (herau) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/models/FoodAPIRestClientUsage.java]: ok !

**Comment by itchix** [comment]
Can I pull? =)

**Pull request merged**